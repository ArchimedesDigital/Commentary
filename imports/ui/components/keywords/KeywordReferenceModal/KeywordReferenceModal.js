import React from 'react';
import { Meteor } from 'meteor/meteor';
import { createContainer } from 'meteor/react-meteor-data';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

// api
import Keywords from '/imports/models/keywords';

// lib
import muiTheme from '/imports/lib/muiTheme';

const KeywordReferenceModal = React.createClass({

	propTypes: {
		visible: React.PropTypes.bool,
		top: React.PropTypes.number,
		left: React.PropTypes.number,
		keyword: React.PropTypes.string,
		close: React.PropTypes.func,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(muiTheme) };
	},

	renderKeywordHTML() {
		let html = '';
		const { keyword } = this.props;

		if (keyword && 'description' in keyword && keyword.description.length) {
			html = `${Utils.trunc(keyword.description, 120)} <a href="/keywords/${keyword.slug}">Read more</a>`;
		}

		return { __html: html };
	},

	render() {
		const self = this;
		const { keyword } = this.props;
		const styles = {
			modal: {
				top: this.props.top,
				left: this.props.left,
			},

		};

		if (!keyword) {
			return null;
		}

		return (
			<div
				className={`reference-modal${(this.props.visible) ?
					' reference-modal-visible' : ''}`}
				style={styles.modal}
			>
				<article className="comment	lemma-comment paper-shadow ">
					<div className="reference-text">
						{keyword ?
							<p
								className="lemma-text"
								dangerouslySetInnerHTML={this.renderKeywordHTML()}
							/>
						: ''}
					</div>

					<i
						className="mdi mdi-close paper-shadow"
						onClick={this.props.close}
					/>
				</article>

			</div>

		);
	},

});

const KeywordReferenceModalContainer = createContainer(({ keywordSlug }) => {
	const query = {
		slug: keywordSlug,
	};
	const handle = Meteor.subscribe('keywords.all', query);
	const keyword = Keywords.findOne(query);

	return {
		keyword,
		ready: handle.ready(),
	};
}, KeywordReferenceModal);

export default KeywordReferenceModalContainer;
