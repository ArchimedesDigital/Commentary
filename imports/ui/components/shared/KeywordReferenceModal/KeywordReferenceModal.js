import React, { Component } from 'react';
import PropTypes from 'prop-types';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import Draggable from 'react-draggable';
import { compose } from 'react-apollo';

// graphql
import { keywordsQuery } from '/imports/graphql/methods/keywords';

// lib
import muiTheme from '/imports/lib/muiTheme';

class KeywordReferenceModal extends Component {

	constructor(props) {
		super(props);
		this.state = {};
		this.renderKeywordHTML = this.renderKeywordHTML.bind(this);

		this.props.keywordsQuery.refetch({
			tenantId: sessionStorage.getItem('tenantId')
		});
	}
	componentWillReceiveProps(nextProps) {
		this.setState({
			keyword: nextProps.keywordsQuery.loading ? [] : nextProps.keywordsQuery.keywords,
			ready: !nextProps.keywordsQuery.loading,
		});
	}
	renderKeywordHTML() {
		let html = '';
		const keyword = this.data.keyword;

		if ('description' in keyword && keyword.description.length) {
			html = `${Utils.trunc(keyword.description, 120)} <a href="/tags/${keyword.slug}">Read more</a>`;
		}

		return { __html: html };
	}

	render() {
		const self = this;
		const { keyword } = this.state;
		const styles = {
			modal: {
				top: this.props.top,
				left: this.props.left,
			},

		};

		return (
			<Draggable>
				<div
					className={`reference-modal${(this.props.visible) ?
						' reference-modal-visible' : ''}`}
					style={styles.modal}
				>
					<article className="comment	lemma-comment paper-shadow ">
						<div className="reference-text">
							{keyword ?
								<p
									className="lemma-text"
									dangerouslySetInnerHTML={this.renderKeywordHTML()}
								/>
							: ''}
						</div>

						<i
							className="mdi mdi-close paper-shadow"
							onClick={this.props.close}
						/>
					</article>

				</div>
			</Draggable>
		);
	}

}

KeywordReferenceModal.propTypes = {
	visible: PropTypes.bool,
	top: PropTypes.number,
	left: PropTypes.number,
	close: PropTypes.func,
	keywordsQuery: PropTypes.object
};

export default compose(keywordsQuery)(KeywordReferenceModal);
