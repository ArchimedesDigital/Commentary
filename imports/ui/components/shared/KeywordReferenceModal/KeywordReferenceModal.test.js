import React from 'react';
import renderer from 'react-test-renderer';

// component:
import KeywordReferenceModal from './KeywordReferenceModal';

describe('KeywordReferenceModal', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<KeywordReferenceModal />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

// TODO Fix ReferenceError: ReactMeteorData is not defined

