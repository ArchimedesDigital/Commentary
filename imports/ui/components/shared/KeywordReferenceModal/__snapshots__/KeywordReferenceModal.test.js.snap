// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordReferenceModal renders correctly 1`] = `
<div
  className="reference-modal"
  style={
    Object {
      "left": undefined,
      "top": undefined,
    }
  }
>
  <article
    className="comment	lemma-comment paper-shadow "
  >
    <div
      className="reference-text"
    >
      
    </div>
    <i
      className="mdi mdi-close paper-shadow"
      onClick={undefined}
    />
  </article>
</div>
`;
